apps/mobile/src/utils/formatters.ts

```diff
@@ -47,3 +47,51 @@ const calculateRatio = (a: Quantity, b: Quantity) => {
   }
   return asDecimal(BigInt(a.amount), a.decimals) / asDecimal(BigInt(b.amount), b.decimals)
 }
+
+/**
+ * Formats a fiat amount with appropriate scaling (k, M, B) and consistent decimal places
+ *
+ * @param amount - String representation of a number (can include commas)
+ * @param options - Optional formatting configuration
+ * @returns Formatted string value
+ *
+ * Examples:
+ * formatFiatAmount("1234.56") => "1.23k"
+ * formatFiatAmount("422838.89") => "422.84k"
+ * formatFiatAmount("1000000") => "1.00M"
+ */
+export const formatFiatAmount = (
+  amount: string,
+  options: {
+    decimalPlaces?: number
+    threshold?: number
+    addSymbol?: boolean
+    currencySymbol?: string
+  } = {},
+): string => {
+  const { decimalPlaces = 2, threshold = 1000, addSymbol = false, currencySymbol = '' } = options
+
+  const numericValue = parseFloat(amount.replace(/,/g, ''))
+
+  if (isNaN(numericValue)) {
+    return amount
+  }
+
+  let formattedValue: string
+  if (numericValue >= 1_000_000_000) {
+    // Billions
+    formattedValue = `${(numericValue / 1_000_000_000).toFixed(decimalPlaces)}B`
+  } else if (numericValue >= 1_000_000) {
+    // Millions
+    formattedValue = `${(numericValue / 1_000_000).toFixed(decimalPlaces)}M`
+  } else if (numericValue >= threshold) {
+    // Thousands
+    formattedValue = `${(numericValue / 1_000).toFixed(decimalPlaces)}k`
+  } else {
+    // Regular numbers
+    formattedValue = numericValue.toFixed(decimalPlaces)
+  }
```

Comment:
Any chance we can use `Intl.NumberFormat` [like the web app](https://github.com/safe-global/safe-wallet-monorepo/blob/dev/apps/web/src/utils/formatNumber.ts) instead of custom logic? The `compact` option does exactly the same.

Reply:
The fiat/token amount formatting should be consistent across clients. I'm not suggesting here to make it a shared package (although this would be ideal) as it's out of scope, but we should re-use the code as much as possible.

Reply:
@Jonathansoufer why can't we use /packages/utils/src/formatNumber.ts
It has a formatAmount|formatCurrency functions and it will format the number the same way it is done in the web app. 
Since it relies on Intl.NumberFormat it will respect the user's region specified in the OS settings.